### PR TITLE
[pangolin] Update to 0.6 and also support osx

### LINKED ISSUE #15996: [Pangolin] update to port <newest commit on master>
Library name: Pangolin (https://github.com/stevenlovegrove/Pangolin)

New version number: Newest Commit on Master

I used vcpkg to install pangolin. The installation itself works perfectly fine, unfortunately the port seems to be old. I come to this conclusion, since I get the following error output when I try to build the DeepSDF project. When I build pangolin manually I do not encounter this problem. The latest release of pangolin dates back to 2016 and since the repository is not updated that frequently I suggest to update the port to the newest commit. 

```
[ 14%] Built target cnpy-static
[ 28%] Built target cnpy
[ 42%] Built target example1
[ 50%] Building CXX object CMakeFiles/PreprocessMesh.dir/src/Utils.cpp.o
[ 57%] Building CXX object CMakeFiles/PreprocessMesh.dir/src/PreprocessMesh.cpp.o
[ 64%] Building CXX object CMakeFiles/PreprocessMesh.dir/src/ShaderProgram.cpp.o
[ 71%] Building CXX object CMakeFiles/SampleVisibleMeshSurface.dir/src/SampleVisibleMeshSurface.cpp.o
[ 78%] Building CXX object CMakeFiles/SampleVisibleMeshSurface.dir/src/Utils.cpp.o
[ 85%] Building CXX object CMakeFiles/SampleVisibleMeshSurface.dir/src/ShaderProgram.cpp.o
In file included from /home/josefkamysek/vcpkg/installed/x64-linux/include/pangolin/gl/glsl.h:39:0,
                 from /home/josefkamysek/DeepLocalShapes/src/ShaderProgram.cpp:3:
/home/josefkamysek/vcpkg/installed/x64-linux/include/pangolin/gl/colour.h: In static member function ‘static pangolin::Colour pangolin::Colour::Hsv(float, float, float, float)’:
/home/josefkamysek/vcpkg/installed/x64-linux/include/pangolin/gl/colour.h:101:30: error: ‘floor’ was not declared in this scope
           const int i = (int)floor(h);
                              ^~~~~
/home/josefkamysek/vcpkg/installed/x64-linux/include/pangolin/gl/colour.h:101:30: note: suggested alternative: ‘float’
           const int i = (int)floor(h);
                              ^~~~~
                              float
/home/josefkamysek/vcpkg/installed/x64-linux/include/pangolin/gl/colour.h: In member function ‘pangolin::Colour pangolin::ColourWheel::GetColourBin(int) const’:
/home/josefkamysek/vcpkg/installed/x64-linux/include/pangolin/gl/colour.h:154:40: error: ‘sqrt’ was not declared in this scope
         float hue = i * 0.5f * (3.0f - sqrt(5.0f));
                                        ^~~~
/home/josefkamysek/vcpkg/installed/x64-linux/include/pangolin/gl/colour.h:154:40: note: suggested alternative: ‘sat’
         float hue = i * 0.5f * (3.0f - sqrt(5.0f));
                                        ^~~~
                                        sat
In file included from /home/josefkamysek/vcpkg/installed/x64-linux/include/pangolin/gl/glsl.h:39:0,
                 from /home/josefkamysek/DeepLocalShapes/src/ShaderProgram.cpp:3:
/home/josefkamysek/vcpkg/installed/x64-linux/include/pangolin/gl/colour.h: In static member function ‘static pangolin::Colour pangolin::Colour::Hsv(float, float, float, float)’:
/home/josefkamysek/vcpkg/installed/x64-linux/include/pangolin/gl/colour.h:101:30: error: ‘floor’ was not declared in this scope
           const int i = (int)floor(h);
                              ^~~~~
/home/josefkamysek/vcpkg/installed/x64-linux/include/pangolin/gl/colour.h:101:30: note: suggested alternative: ‘float’
           const int i = (int)floor(h);
                              ^~~~~
                              float
/home/josefkamysek/vcpkg/installed/x64-linux/include/pangolin/gl/colour.h: In member function ‘pangolin::Colour pangolin::ColourWheel::GetColourBin(int) const’:
/home/josefkamysek/vcpkg/installed/x64-linux/include/pangolin/gl/colour.h:154:40: error: ‘sqrt’ was not declared in this scope
         float hue = i * 0.5f * (3.0f - sqrt(5.0f));
                                        ^~~~
/home/josefkamysek/vcpkg/installed/x64-linux/include/pangolin/gl/colour.h:154:40: note: suggested alternative: ‘sat’
         float hue = i * 0.5f * (3.0f - sqrt(5.0f));
                                        ^~~~
                                        sat
/home/josefkamysek/DeepLocalShapes/src/ShaderProgram.cpp: In function ‘pangolin::GlSlProgram GetShaderProgram()’:
/home/josefkamysek/DeepLocalShapes/src/ShaderProgram.cpp:146:31: error: ‘GlSlAnnotatedShader’ is not a member of ‘pangolin’
   program.AddShader(pangolin::GlSlAnnotatedShader, shaderText);
                               ^~~~~~~~~~~~~~~~~~~
/home/josefkamysek/DeepLocalShapes/src/ShaderProgram.cpp:146:31: note: suggested alternative: ‘GlSlComputeShader’
   program.AddShader(pangolin::GlSlAnnotatedShader, shaderText);
                               ^~~~~~~~~~~~~~~~~~~
                               GlSlComputeShader
/home/josefkamysek/DeepLocalShapes/src/ShaderProgram.cpp: In function ‘pangolin::GlSlProgram GetShaderProgram()’:
/home/josefkamysek/DeepLocalShapes/src/ShaderProgram.cpp:146:31: error: ‘GlSlAnnotatedShader’ is not a member of ‘pangolin’
   program.AddShader(pangolin::GlSlAnnotatedShader, shaderText);
                               ^~~~~~~~~~~~~~~~~~~
/home/josefkamysek/DeepLocalShapes/src/ShaderProgram.cpp:146:31: note: suggested alternative: ‘GlSlComputeShader’
   program.AddShader(pangolin::GlSlAnnotatedShader, shaderText);
```

### PR DESCRIPTION
**Pull request by NancyLi1013**
**Describe the pull request**

- #### What does your PR fix?  
  Fixes #15996 
           #15834

1. Update to the latest release 0.6.
2. Add support on osx.
3. Add test, examples and tools as features.

4. Remove the deprecated patch `deprecated_constants.patch`, since it has been merged to the version.

5. Fix build error on Visual Studio 2019.
upstream issue https://github.com/stevenlovegrove/Pangolin/issues/654
The solution is from the user and I have confirmed with upstream owner.

Note: All features have passed with the following triplets:

- x86-windows
- x64-windows-static

### CODE REVIEW

File: versions/p-/pangolin.json [hunk 1]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "ed55513cd210d04ee2d3fbda04813641d3fc7da8",
```
Comment (NancyLi1013) [inline versions/p-/pangolin.json:4]:
```suggestion
      "git-tree": "84a730418438c11d401ac4f48842b30547b2c2d5",
```

File: versions/p-/pangolin.json [hunk 2]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "84a730418438c11d401ac4f48842b30547b2c2d5",
```
Comment (NancyLi1013) [inline versions/p-/pangolin.json:4]:
```suggestion
      "git-tree": "cba2fbfee6c21df4d91853baabca16785e15ddb5",
```

File: ports/pangolin/portfile.cmake [hunk 3]
```diff
@@ -1,34 +1,46 @@
...
-vcpkg_configure_cmake(
-    SOURCE_PATH ${SOURCE_PATH}
-    PREFER_NINJA
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS
         -DBUILD_EXTERN_GLEW=OFF
         -DBUILD_EXTERN_LIBPNG=OFF
         -DBUILD_EXTERN_LIBJPEG=OFF
-        -DCMAKE_DISABLE_FIND_PACKAGE_PythonLibs=ON
```
Comment (NancyLi1013) [inline ports/pangolin/portfile.cmake:31]: PythonLibs isn't required on current version.

**Review by JackBoosY — APPROVED** [review]

**Review by JackBoosY — CHANGES_REQUESTED** [review]
```
CMake Warning at /Users/vagrant/Data/installed/x64-osx/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:472 (message):
  The following variables are not used in CMakeLists.txt:

      MSVC_USE_STATIC_CRT

  Please recheck them and remove the unnecessary options from the
  `vcpkg_cmake_configure` call.

  If these options should still be passed for whatever reason, please use the
```

**Review by JackBoosY — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for your help with the update!

**Pull request merged**